qiskit/providers/__init__.py

```diff
@@ -47,15 +47,9 @@
    JobTimeoutError
 """
 
-import pkgutil
-
 from .basebackend import BaseBackend
 from .baseprovider import BaseProvider
 from .basejob import BaseJob
 from .exceptions import (JobError, JobTimeoutError, QiskitBackendNotFoundError,
                          BackendPropertyError, BackendConfigurationError)
 from .jobstatus import JobStatus
-
-
-# Allow extending this namespace.
-__path__ = pkgutil.extend_path(__path__, __name__)
```

Comment:
The two approaches are incompatible for the packages we add the meta-importer for. This is because the meta importer will be used for package lookups and any namespace packages won't be found. I haven't tested trying to use them together for non-qiskit qiskit extensions, but I don't think it's something worth doing as I imagine it will get pretty hairy between old element versions more than it would be useful for external things extending the namespace.